.changeset/silent-parrots-pay.md

```
@@ -0,0 +1,6 @@
+---
+'@mastra/server': patch
+'@mastra/core': patch
+---
+
+Added public read-only thread query methods (`queryThreads`, `queryThreadById`, `queryThreadMessages`) on `AgentController`. These read directly from storage without constructing a Session, so callers can list threads or fetch messages without triggering the workspace/sandbox initialization that `createSession` performs as a side effect.
```

markdownlint-cli2 (0.23.2)

[warning] First line in a file should be a top-level heading

(MD041, first-line-heading, first-line-h1)